tron/core/action.py

```diff
@@ -117,37 +117,42 @@ def serialize_namedtuple(obj):
         try:
             return json.dumps(
                 {
-                    "command": state_data["command"],
-                    "cpus": state_data["cpus"],
-                    "mem": state_data["mem"],
-                    "disk": state_data["disk"],
-                    "cap_add": state_data["cap_add"],
-                    "cap_drop": state_data["cap_drop"],
+                    "command": state_data.get("command"),
+                    "cpus": state_data.get("cpus"),
+                    "mem": state_data.get("mem"),
+                    "disk": state_data.get("disk"),
+                    "cap_add": state_data.get("cap_add", []),
+                    "cap_drop": state_data.get("cap_drop", []),
```

Comment:
we have a default if cap_drop isn't provided in task_proc - but i think to take advantage of that we'd have to not pass through a value if we didn't get one, which would probably be a bit tricky

i sorta wonder if we should instead default to re-adding the default cap drops here

although: i'm kinda surpised that there's stuff missing these to begin with

Reply:
IIRC I have seen a job that was either missing cap_add or cap_drop. There is a lot of ancient data in the db that had features no longer supported . I see in actioncommandconfig theyre set to the config if it exists or empty list
```
            cap_add=config.cap_add or [],
            cap_drop=config.cap_drop or [],
```